src/System.Linq/tests/ShortCircuitingTests.cs

```diff
@@ -111,6 +111,94 @@ public void MinNullableSingleDoesntCheckAll()
             Assert.Equal(5, tracker.Moves);
         }
 
+        [Fact]
+        public void MinDoubleDoesntCheckAllStartLeadingWithNaN()
+        {
+            var tracker = new TrackingEnumerable(10);
+            var source = tracker.Select(i => i == 1 ? double.NaN : (double)i);
```

Comment:
Please only use `var` when the type is spelled out on the right-hand side.  This should be `IEnumerable<double>`.